File: 백준/Silver/10828. 스택/스택.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.Deque;
import java.util.LinkedList;
import java.util.StringTokenizer;

public class Main {

	public static void main(String[] args) throws IOException {
		
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
		String str;
		StringTokenizer st;
		
		int n, i, num;
		String command;
		Deque<Integer> stack = new LinkedList<>();
		
		str = br.readLine();
		n = Integer.parseInt(str);
		
		for (i = 0; i < n; i++) {
			
			str = br.readLine();
			st = new StringTokenizer(str, " ");
			command = st.nextToken();
			
			if (command.equals("push")) {
				num = Integer.parseInt(st.nextToken());
				stack.add(num);
			}
			else if (command.equals("pop")) {
				if (stack.isEmpty()) {
					num = -1;
				}
				else {
					num = stack.pollLast();
				}
				bw.write(Integer.toString(num) + "\n");
			}
			else if (command.equals("size")) {
				num = stack.size();
				bw.write(Integer.toString(num) + "\n");
			}
			else if (command.equals("empty")) {
				if (stack.isEmpty()) {
					num = 1;
				}
				else {
					num = 0;
				}
				bw.write(Integer.toString(num) + "\n");
			}
			else if (command.equals("top")) {
				if (stack.isEmpty()) {
					num = -1;
				}
				else {
					num = stack.peekLast();
				}
				bw.write(Integer.toString(num) + "\n");
			}
			
		}
		
		bw.flush();
		bw.close();

	}

}

File: 백준/Silver/10828. 스택/스택.c
#include <stdio.h>
#define MAX_STACK_SIZE 10000
#define MAX_STRING_SIZE 6

typedef struct Stack
{
	int head;
	int elements[MAX_STACK_SIZE];
} Stack;

int size(Stack* stack)
{
	return stack->head + 1;
}

int empty(Stack* stack)
{
    return !size(stack);
}

void push(Stack* stack, int num)
{
	stack->head++;
	stack->elements[stack->head] = num;
}

int pop(Stack* stack)
{
	if (empty(stack))
	{
		return -1;
	}
	stack->head--;
	return stack->elements[stack->head + 1];
}

int top(Stack* stack)
{
	return empty(stack) ? -1 : stack->elements[stack->head];
}

int main(void)
{
	int num;
	int test_case;
	char str[MAX_STRING_SIZE];
	Stack stack;

	stack.head = -1;
	scanf("%d", &test_case);
	for (int i = 0; i < test_case; i++)
	{
		scanf("%s", str);
		if (str[1] == 'u')
		{
			scanf("%d", &num);
			push(&stack, num);
		}
		else if (str[0] == 't')
		{
			printf("%d\n", top(&stack));
		}
		else if (str[0] == 's')
		{
			printf("%d\n", size(&stack));
		}
		else if (str[0] == 'e')
		{
			printf("%d\n", empty(&stack));
		}
		else
		{
			printf("%d\n", pop(&stack));
		}
	}
	return 0;
}